src/test/java/com/test/fibonacci/FibonacciControllerTests.java

```diff
@@ -0,0 +1,99 @@
+package com.test.fibonacci;
+
+import com.test.fibonacci.businesslogic.entity.Response;
+import com.test.fibonacci.controller.FibonacciController;
+import com.test.fibonacci.handler.FibonacciHandler;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.json.AutoConfigureJsonTesters;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.json.JacksonTester;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import java.util.ArrayList;
+
+import static com.test.fibonacci.util.UserMessages.BAD_VALUE;
+import static com.test.fibonacci.util.UserMessages.NEGATIVE_USER_ID_MESSAGE;
+import static com.test.fibonacci.util.UserMessages.NULL_ERROR_MESSAGE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+
+@AutoConfigureJsonTesters
+@WebMvcTest(FibonacciController.class)
+public class FibonacciControllerTests {
+
+    @Autowired private MockMvc mvc;
+    @Autowired private JacksonTester<Response> jsonResponse;
+    @MockBean private FibonacciHandler fibonacciHandler;
+
+    @Test
+    public void negativeUserId() throws Exception {
```

Comment:
In general, e recomandat ca testele unitare sa aiba o denumire ce poate fi descrisa ca `UnitOfWork_StateUnderTest_ExpectedBehavior` sau `subject_when_then` ex:
`GET_sequence_when_user_id_is_negative_returns_a_specific_message`.